src/rebar3_coveralls.erl

```diff
@@ -153,6 +156,44 @@ do_coveralls(ConvertAndSend, Get, GetLocal, MaybeSkip, Task) ->
     _ -> MaybeSkip()
   end.
 
+git_trim(Str) ->
+  re:replace(Str, "\\s+", "", [global, {return, binary}]).
+
+collect_git_info(Report) ->
+  case rebar_utils:sh("git rev-parse --verify HEAD", [return_on_error]) of
+    {ok, Id} ->
+      Report#{git => collect_git_details(git_trim(Id))};
+    _ ->
+      %% git not available
+      Report
+  end.
+
+collect_git_details(Id) ->
+  {ok, Details} = rebar_utils:sh("git show -s --pretty=format:'%aN%n%aE%n%cN%n%cE%n%s'", []),
+  Info = re:split(Details, "\n", [{return, binary}]),
+  Head = maps:from_list(
+           lists:zip(
+             ['author_name', 'author_email', 'committer_name', 'committer_email', 'message'], Info)),
+  collect_git_branch_details(#{head => Head#{id => Id}}).
+
+collect_git_branch_details(Git) ->
+  {ok, Branch} = rebar_utils:sh("git rev-parse --abbrev-ref HEAD", []),
```

Comment:
If this is specifically needed by GitHub actions, can't we use the `GITHUB_SHA` variable or any other appropriate variable as specified at `https://docs.github.com/en/actions/reference/environment-variables`?

Reply:
Also, by looking at the Erlang LS / Coveralls interaction (https://coveralls.io/github/erlang-ls/erlang_ls) it looks like the major issue with it is that somehow coverage for Pull Requests is correctly reported, but things do not work for `master/main`.

Reply:
> If this is specifically needed by GitHub actions, can't we use the `GITHUB_SHA` variable or any other appropriate variable as specified at `https://docs.github.com/en/actions/reference/environment-variables`?

The env variables only contain the commit SHA, but we need more than that. All fields ine git key need to be filled.

Reply:
> Also, by looking at the Erlang LS / Coveralls interaction (https://coveralls.io/github/erlang-ls/erlang_ls) it looks like the major issue with it is that somehow coverage for Pull Requests is correctly reported, but things do not work for `master/main`.

Erlang LS is not using the changes from this PR, hence the wrong reporting.
Also, even when/if this PR is accepted, the GH Action in Erlang LS would need to be adjusted (see the README.md changes in the PR).